core/src/ldml/ldml_processor.cpp

```diff
@@ -294,25 +263,33 @@ ldml_processor::process_event(
       // Now process transforms
       std::u16string outputString;
       // Process the transforms
-      const size_t matchedContext = transforms.matchContext(ctxt, outputString);
-
-      if (matchedContext > 0) {
-        // Found something.
-        // Now, clear out the old context
-        for (size_t i = 0; i < matchedContext; i++) {
-          state->context().pop_back();  // Pop off last
-          auto deletedChar = ctxt[ctxt.size() - i - 1][0];
-          state->actions().push_backspace(KM_KBP_BT_CHAR, deletedChar);  // Cause prior char to be removed
+      if (!!transforms) {
+
+        // TODO-LDML: unroll ctxt into a str
+        std::u16string ctxtstr;
+        for (size_t i = 0; i < ctxt.size(); i++) {
+          ctxtstr.append(ctxt[i]);
         }
-        // Now, add in the updated text
-        const std::u32string outstr32 = kmx::u16string_to_u32string(outputString);
-        for (size_t i = 0; i < outstr32.length(); i++) {
-          state->context().push_character(outstr32[i]);
-          state->actions().push_character(outstr32[i]);
+        const size_t matchedContext = transforms->apply(ctxtstr, outputString);
+
+        if (matchedContext > 0) {
+          // Found something.
+          // Now, clear out the old context
+          for (size_t i = 0; i < matchedContext; i++) {
+            state->context().pop_back();  // Pop off last
+            auto deletedChar = ctxt[ctxt.size() - i - 1][0];
+            state->actions().push_backspace(KM_KBP_BT_CHAR, deletedChar);  // Cause prior char to be removed
+          }
+          // Now, add in the updated text
+          const std::u32string outstr32 = kmx::u16string_to_u32string(outputString);
+          for (size_t i = 0; i < outstr32.length(); i++) {
+            state->context().push_character(outstr32[i]);
+            state->actions().push_character(outstr32[i]);
+          }
+          // Add a marker so we don't retransform this text.
+          state->context().push_marker(0x0);
+          state->actions().push_marker(0x0);
```

Comment:
I'm a little puzzled by this -- why do we need to do this?

Reply:
aha! left over from 'old transforms', will remove